service/src/main/java/com/epam/esm/service/service/impl/TagGiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.epam.esm.service.service.impl;
+
+import com.epam.esm.persistence.entity.Tag;
+import com.epam.esm.persistence.model.TagGiftCertificateId;
+import com.epam.esm.persistence.repository.TagGiftCertificateRepository;
+import com.epam.esm.persistence.repository.TagRepository;
+import com.epam.esm.service.dto.TagDto;
+import com.epam.esm.service.service.TagGiftCertificateService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class TagGiftCertificateServiceImpl implements TagGiftCertificateService {
+
+    private final ModelMapper modelMapper;
+    private final TagRepository tagRepository;
+    private final TagGiftCertificateRepository tagCertificateRepository;
+
+    @Autowired
+    public TagGiftCertificateServiceImpl(ModelMapper modelMapper, TagRepository tagRepository, TagGiftCertificateRepository tagCertificateRepository) {
+        this.modelMapper = modelMapper;
+        this.tagRepository = tagRepository;
+        this.tagCertificateRepository = tagCertificateRepository;
+    }
+
+    @Override
+    public void saveCertificateTags(Long certificateId, Set<Tag> certificateTags) {
+        Set<Tag> foundTags = getFoundTags(certificateTags);
+        Set<Long> foundTagsIds = foundTags.stream().map(Tag::getId).collect(Collectors.toSet());
```

Comment:
Streams become more readable if you transfer each operation to a new line

Reply:
fixed